graphics.cpp

```diff
@@ -72,3 +72,59 @@ GLubyte *Texture::data() const {
 
 	return texture_data;
 }
+
+Point operator+(const Point &l, const Point &r) {
+	double x = get<X>(l) + get<X>(r);
+	double y = get<Y>(l) + get<Y>(r);
+
+	return Point(x, y);
+}
+
+Point operator+=(Point &l, const Point &r) {
```

Comment:
An actual nitpick, if these operators stick around: assignment operators should return a reference. It eliminates the potential for a useless copy and is just generally the idiomatic way to do it.